### PR TITLE
Add TransactionHandler Abstract Base Class to Python SDK

### PR DESCRIPTION
**Pull request by agunde406**
The other SDKs have an interface defined for TransactionHandlers, but python did not. This PR adds an Abstract Base Class that all TransactionHandlers should inherit from. It also updates the ValidatorRegistryTransactionHandler to expect context instead of state.

### CODE REVIEW

File: sdk/python/sawtooth_sdk/processor/handler.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    @abc.abstractproperty
+    def family_name(self):
+        """
+        FamilyName should return the name of the transaction family that this
+        handler can process. Eg., "intkey"
+        """
+        pass
+
+    @abc.abstractproperty
+    def family_versions(self):
```
Comment (nick-drozd) [inline sdk/python/sawtooth_sdk/processor/handler.py]: Is this a string or a list of strings?

Comment (agunde406) [reply sdk/python/sawtooth_sdk/processor/handler.py]: Good catch. I updated it say it is a list. I think the other SDKs have this as just a string. We should create a Story to update go and javascript to a list and make sure the filtering works correctly.

File: sdk/python/sawtooth_sdk/processor/handler.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import abc
+
+
+class TransactionHandler(object, metaclass=abc.ABCMeta):
+    """
+    TransactionHandler is the Abstract Base Class that defines the business
+    logic for a new transaction family.
+
+    The family_name, family_version, and namespaces properties are
```
Comment (nick-drozd) [inline sdk/python/sawtooth_sdk/processor/handler.py]: family_versions

File: sdk/python/sawtooth_sdk/processor/handler.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    used by the processor to route processing requests to the handler.
+    """
+
+    @abc.abstractproperty
+    def family_name(self):
+        """
+        family_Name should return the name of the transaction family that this
```
Comment (nick-drozd) [inline sdk/python/sawtooth_sdk/processor/handler.py]: family_name

File: sdk/python/sawtooth_sdk/processor/handler.py (hunk 2 above)
Comment (nick-drozd) [inline sdk/python/sawtooth_sdk/processor/handler.py]: family_versions

File: sdk/python/sawtooth_sdk/processor/handler.py (hunk 3 above)
Comment (nick-drozd) [inline sdk/python/sawtooth_sdk/processor/handler.py]: family_name

**Pull request merged**